### PR TITLE
chore: improve docker testing

### PR DESCRIPTION
**Pull request by nrwiersma**
## What is the problem I am trying to address?

This PR fixes the docker e2e tests that are currently broken and additionally adds mod cache mounting to make testing in docker faster.

## How is the fix applied?

The mod cache location is determined from the go environment and volume mounted into test containers.
As the docker e2e testing seems to not have been updated when e2e tests changed, I fix that as well.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -37,7 +39,9 @@ services:
       dockerfile: Dockerfile.test
       args:
         GOLANG_VERSION: 1.19
-    command: ["./scripts/test_e2e.sh"]
+    command: "/bin/sh -c 'cd e2etests && go test --tags e2etests'"
+    volumes:
+      - ${GO_MODCACHE}:/go/pkg/mod
```
Comment (manugupt1) [inline docker-compose.yml:44]: Won't this fail if GO_MODCACHE is not set?

Comment (nrwiersma) [reply docker-compose.yml:44]: No, you get a warning that it is not set, and then it just downloads the mods.

Comment (manugupt1) [reply docker-compose.yml:44]: Hmm! I tried locally. I am getting the following error if GO_MODCACHE is not set or empty

```
WARNING: using --exit-code-from implies --abort-on-container-exit
Creating athense2e_teste2e_1 ... error

ERROR: for athense2e_teste2e_1  Cannot create container for service teste2e: create .: volume name is too short, names should be at least two alphanumeric characters

ERROR: for teste2e  Cannot create container for service teste2e: create .: volume name is too short, names should be at least two alphanumeric characters
ERROR: Encountered errors while bringing up the project.
make: *** [Makefile:76: test-e2e-docker] Error 1
➜  athens git:(fix-docker-e2e) ✗ 
```

Comment (nrwiersma) [reply docker-compose.yml:44]: Fair

**Pull request closed without merging**